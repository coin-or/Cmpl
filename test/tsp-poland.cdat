%start < Bielsko-Biala  >

%cities set <
Bielsko-Biala  
Jastrzebie-Zdro
Krosno         
NowySacz       
Przemysl       
Bytom          
Chorzow        
Czestochowa    
DabrowaGornicz 
Gliwice        
Jaworzno       
JeleniaGora    
Katowice       
Kielce         
Krakow         
Myslowice      
Opole          
Piekaryslaskie 
Rudaslaska     
Rybnik         
Rzeszow        
SiemianowiceSl 
Sosnowiec      
Swietochlowice 
Tarnobrzeg     
Tarnow         
Tychy          
Walbrzych      
Zabrze         
Zamosc         
Zory           
Chelm          
Kalisz         
Legnica        
Leszno         
Lodz           
Lublin         
PiotrkowTrybun 
Radom          
Skierniewice   
Wroclaw        
ZielonaGora    
BialaPodlaska  
GorzowWielkopo 
Konin          
Plock          
Poznan         
Siedlce        
Warszawa       
Wloclawek      
Bialystok      
Bydgoszcz      
Grudziadz      
lomZa          
Olsztyn        
Ostroleka      
Swinoujscie    
Szczecin       
Torun          
Elblag         
Gdansk         
Gdynia         
Koszalin       
Slupsk         
Sopot          
Suwalki        
>

%d[cities,cities] <
100000	37	195	121	267	60	54	109	58	58	45	265	50	164	69	48	124	63	53	47	213	56	52	56	206	140	35	224	57	316	36	346	228	255	287	217	294	181	231	251	204	342	377	419	273	307	323	347	301	315	465	374	408	428	452	403	563	507	358	485	505	526	524	536	514	544
37	100000	230	157	301	51	47	102	60	37	57	228	47	180	97	52	93	55	42	17	245	52	54	48	231	172	35	187	41	344	14	371	206	220	256	210	316	179	244	250	170	309	393	387	254	300	296	359	305	303	475	355	393	433	446	406	532	476	342	473	490	510	499	514	499	548
195	230	100000	76	74	216	212	225	198	230	186	449	206	155	138	198	294	215	217	234	45	207	199	215	100	67	205	408	223	159	223	203	348	430	438	281	182	241	196	279	371	506	279	567	374	350	454	281	287	379	396	463	470	389	465	379	699	649	431	525	561	583	630	623	572	496
121	157	76	100000	150	151	146	173	134	163	121	381	140	140	75	133	230	151	150	163	105	143	134	149	128	49	136	340	157	220	151	259	302	366	384	252	223	211	201	265	309	447	318	515	337	334	407	306	290	356	426	432	450	407	463	390	653	600	406	514	545	566	595	595	555	520
267	301	74	150	100000	282	279	283	264	298	254	515	274	194	206	266	359	281	284	304	63	274	266	282	118	131	274	475	291	111	293	158	397	493	494	319	162	283	213	306	433	564	252	619	417	375	502	269	298	410	374	498	496	380	472	377	745	697	462	540	581	603	666	653	592	479
60	51	216	151	282	100000	6	52	19	19	31	233	11	136	78	19	79	4	9	38	222	8	17	3	197	151	25	192	11	310	37	333	169	215	236	161	275	128	197	199	158	297	348	368	213	251	267	310	255	256	425	315	349	382	396	355	509	455	298	426	446	466	465	476	455	497
54	47	212	146	279	6	100000	57	17	20	26	237	6	136	74	14	83	9	6	36	219	6	13	3	196	148	19	196	12	309	33	332	175	220	241	165	275	133	198	203	163	302	349	374	219	256	273	312	258	262	428	321	354	385	401	359	515	460	304	432	451	472	471	482	461	501
109	102	225	173	283	52	57	100000	52	66	68	238	59	108	100	61	85	48	61	88	221	54	58	55	182	158	75	200	61	292	89	308	130	211	214	109	246	77	158	148	151	281	310	343	169	199	234	267	206	206	379	269	299	333	344	305	480	427	250	376	396	417	426	432	406	447
58	60	198	134	264	19	17	52	100000	37	16	252	13	121	61	10	96	17	23	52	203	11	7	19	179	133	26	211	29	293	46	316	178	233	250	159	259	124	184	194	175	313	334	382	220	250	279	298	247	258	415	321	352	374	394	348	522	468	302	428	449	470	475	483	458	489
58	37	230	163	298	19	20	66	37	100000	44	219	25	155	93	33	68	23	14	22	239	26	33	18	216	167	28	178	8	329	26	352	171	204	230	173	294	143	215	214	149	288	367	362	218	263	266	329	271	265	443	320	356	398	409	371	505	450	305	436	453	473	466	480	462	513
45	57	186	121	254	31	26	68	16	44	100000	263	20	123	49	12	109	31	31	53	195	22	14	29	175	123	23	222	37	289	45	314	194	246	266	173	259	137	188	206	189	327	337	398	236	263	295	304	256	273	422	337	367	383	407	358	538	484	317	442	464	485	491	499	473	499
265	228	449	381	515	233	237	238	252	219	263	100000	243	345	311	251	157	235	232	219	454	242	250	234	421	385	245	41	226	531	231	543	190	45	123	277	480	281	384	328	94	116	527	207	228	330	186	475	395	302	565	292	354	503	455	469	351	292	307	441	432	446	366	407	437	602
50	47	206	140	274	11	6	59	13	25	20	243	100000	134	69	8	89	13	11	38	214	6	8	9	192	143	16	202	17	305	33	329	179	226	247	168	272	134	197	205	169	308	347	379	224	258	278	311	258	265	427	325	358	386	403	359	521	466	307	435	455	475	476	487	464	501
164	180	155	140	194	136	136	108	121	155	123	345	134	100000	104	128	193	132	142	172	133	130	127	137	80	99	145	307	147	185	166	200	204	316	304	127	140	89	68	127	255	376	214	426	223	197	308	184	151	225	303	308	317	273	323	253	551	503	277	376	409	431	476	468	420	388
69	97	138	75	206	78	74	100	61	93	49	311	69	104	100000	60	157	78	79	99	148	70	62	77	137	75	68	270	86	248	88	277	229	294	309	190	227	149	172	212	236	373	312	441	267	277	334	287	251	293	407	365	388	376	415	354	579	526	341	458	484	506	524	528	494	491
48	52	198	133	266	19	14	61	10	33	12	251	8	128	60	100000	97	19	19	45	206	11	3	17	185	134	17	210	26	298	38	322	184	234	254	169	266	134	192	204	177	315	342	386	228	259	284	307	256	267	424	329	360	383	404	358	527	473	311	437	458	478	481	491	467	499
124	93	294	230	359	79	83	85	96	68	109	157	89	193	157	97	100000	79	79	77	297	87	95	80	265	228	96	117	73	377	88	393	123	137	164	161	331	147	241	211	81	220	391	295	174	242	205	345	276	234	451	273	318	398	387	367	440	384	266	402	413	432	408	427	421	509
63	55	215	151	281	4	9	48	17	23	31	235	13	132	78	19	79	100000	12	42	220	9	17	7	195	150	28	194	15	308	41	330	166	216	235	157	272	125	193	195	158	296	344	367	211	247	265	306	251	253	421	312	345	378	393	351	508	454	295	423	442	463	463	474	452	493
53	42	217	150	284	9	6	61	23	14	31	232	11	142	79	19	79	12	100000	30	225	12	18	6	202	153	18	191	7	315	28	338	175	216	239	169	281	137	204	208	160	299	355	371	220	259	272	318	263	264	433	322	356	390	405	363	514	459	306	435	453	474	471	483	462	506
47	17	234	163	304	38	36	88	52	22	53	219	38	172	99	45	77	42	30	100000	247	42	46	36	229	174	31	177	28	342	12	367	189	208	241	195	310	165	234	236	156	295	385	373	237	284	280	348	293	286	463	338	377	420	430	392	517	461	325	457	473	493	482	497	482	535
213	245	45	105	63	222	219	221	203	239	195	454	214	133	148	206	297	220	225	247	100000	214	206	222	63	73	216	414	231	117	236	159	335	431	431	260	137	222	161	251	371	501	234	557	356	321	440	238	251	353	352	440	442	347	428	339	683	635	406	492	530	552	608	597	541	453
56	52	207	143	274	8	6	54	11	26	22	242	6	130	70	11	87	9	12	42	214	100000	8	8	190	143	21	201	18	303	38	326	175	224	244	162	269	129	193	199	167	305	343	376	218	253	274	307	253	259	422	320	352	381	398	354	517	462	302	429	449	470	471	481	459	496
52	54	199	134	266	17	13	58	7	33	14	250	8	127	62	3	95	17	18	46	206	8	100000	16	184	135	20	209	25	297	40	321	181	232	252	165	264	131	190	201	175	313	340	384	224	256	282	305	253	264	421	326	357	381	400	355	525	470	307	433	454	475	478	488	464	496
56	48	215	149	282	3	3	55	19	18	29	234	9	137	77	17	80	7	6	36	222	8	16	100000	198	151	22	193	10	311	34	334	172	217	238	164	276	132	199	202	160	299	349	370	217	254	270	312	258	260	428	318	352	385	400	358	512	457	301	430	449	469	468	479	458	500
206	231	100	128	118	197	196	182	179	216	175	421	192	80	137	185	265	195	202	229	63	190	184	198	100000	80	198	381	208	113	220	141	284	393	384	202	96	167	98	189	332	456	190	505	300	259	387	183	188	292	301	379	379	290	366	278	628	582	345	429	467	489	548	535	478	399
140	172	67	49	131	151	148	158	133	167	123	385	143	99	75	134	228	150	153	174	73	143	135	151	80	100000	143	344	160	180	163	216	282	364	371	220	175	179	154	225	305	439	269	500	310	295	388	257	245	321	377	402	414	359	420	343	633	583	372	474	507	529	567	564	518	471
35	35	205	136	274	25	19	75	26	28	23	245	16	145	68	17	96	28	18	31	216	21	20	22	198	143	100000	204	24	311	22	337	193	231	257	184	281	150	209	220	176	315	358	389	238	274	291	324	273	281	441	340	373	400	419	374	532	477	323	451	470	491	490	501	480	516
224	187	408	340	475	192	196	200	211	178	222	41	202	307	270	210	117	194	191	177	414	201	209	193	381	344	204	100000	184	492	189	506	168	48	124	248	443	248	349	299	65	140	495	230	211	308	187	444	367	285	539	287	346	479	440	445	377	318	297	434	430	445	380	415	435	582
57	41	223	157	291	11	12	61	29	8	37	226	17	147	86	26	73	15	7	28	231	18	25	10	208	160	24	184	100000	321	28	344	171	209	233	168	286	138	208	208	153	292	359	365	217	258	267	321	265	262	436	318	354	392	404	365	508	453	303	433	451	471	467	479	460	507
316	344	159	220	111	310	309	292	293	329	289	531	305	185	248	298	377	308	315	342	117	303	297	311	113	180	311	492	321	100000	333	48	380	500	483	289	75	262	165	259	440	557	146	597	386	320	477	176	228	360	268	450	435	285	390	287	709	668	410	465	511	531	616	593	522	375
36	14	223	151	293	37	33	89	46	26	45	231	33	166	88	38	88	41	28	12	236	38	40	34	220	163	22	189	28	333	100000	358	196	220	252	197	303	165	230	236	168	307	380	383	244	287	290	345	291	291	461	345	382	418	433	392	528	472	331	461	479	499	491	505	488	534
346	371	203	259	158	333	332	308	316	352	314	543	329	200	277	322	393	330	338	367	159	326	321	334	141	216	337	506	344	48	358	100000	381	510	486	287	64	266	164	250	450	559	103	593	381	304	473	143	208	346	224	435	414	246	358	253	697	659	395	436	484	504	598	571	495	330
228	206	348	302	397	169	175	130	178	171	194	190	179	204	229	184	123	166	175	189	335	175	181	172	284	282	193	168	171	380	196	381	100000	147	104	96	317	118	218	143	103	178	347	222	51	141	106	292	208	119	376	150	196	313	276	279	351	301	144	281	290	309	297	309	298	415
255	220	430	366	493	215	220	211	233	204	246	45	226	316	294	234	137	216	216	208	431	224	232	217	393	364	231	48	209	500	220	510	147	100000	80	237	446	245	349	287	61	94	489	182	183	285	143	435	354	257	523	248	309	459	410	425	329	271	262	396	388	403	332	369	394	558
287	256	438	384	494	236	241	214	250	230	266	123	247	304	309	254	164	235	239	241	431	244	252	238	384	371	257	124	233	483	252	486	104	80	100000	200	421	221	322	245	89	74	450	132	121	227	64	393	308	192	467	170	233	400	338	365	276	220	188	318	309	324	258	291	314	492
217	210	281	252	319	161	165	109	159	173	173	277	168	127	190	169	161	157	169	195	260	162	165	164	202	220	184	248	168	289	197	287	96	237	200	100000	223	42	124	52	183	274	253	309	99	90	189	199	119	104	293	182	198	237	236	206	426	381	153	269	295	316	349	344	305	348
294	316	182	223	162	275	275	246	259	294	259	480	272	140	227	266	331	272	281	310	137	269	264	276	96	175	281	443	286	75	303	64	317	446	421	223	100000	202	99	188	387	495	97	530	318	246	410	108	153	287	215	376	360	219	317	217	637	597	337	391	436	457	542	518	447	318
181	179	241	211	283	128	133	77	124	143	137	281	134	89	149	134	147	125	137	165	222	129	131	132	167	179	150	248	138	262	165	266	118	245	221	42	202	100000	103	71	187	295	247	339	134	128	220	199	130	145	305	223	240	256	270	227	461	414	195	309	336	357	389	385	346	370
231	244	196	201	213	197	198	158	184	215	188	384	197	68	172	192	241	193	204	234	161	193	190	199	98	154	209	349	208	165	230	164	218	349	322	124	99	103	100000	96	290	396	152	432	221	163	312	116	92	200	235	289	283	207	269	189	545	502	251	331	369	391	457	440	380	321
251	250	279	265	306	199	203	148	194	214	206	328	205	127	212	204	211	195	208	236	251	199	201	202	189	225	220	299	208	259	236	250	143	287	245	52	188	71	96	100000	235	318	205	344	132	71	224	149	68	104	242	193	191	188	203	158	450	409	156	250	283	304	361	347	293	300
204	170	371	309	433	158	163	151	175	149	189	94	169	255	236	177	81	158	160	156	371	167	175	160	332	305	176	65	153	440	168	450	103	61	89	183	387	187	290	235	100000	139	433	218	149	243	143	381	302	221	474	232	288	413	377	380	364	307	237	375	376	393	345	373	382	517
342	309	506	447	564	297	302	281	313	288	327	116	308	376	373	315	220	296	299	295	501	305	313	299	456	439	315	140	292	557	307	559	178	94	74	274	495	295	396	318	139	100000	522	91	191	294	110	465	380	257	535	215	280	466	393	431	237	178	243	361	342	353	255	301	345	552
377	393	279	318	252	348	349	310	334	367	337	527	347	214	312	342	391	344	355	385	234	343	340	349	190	269	358	495	359	146	380	103	347	489	450	253	97	247	152	205	433	522	100000	541	333	240	425	60	144	283	122	367	335	145	263	157	630	598	325	344	394	413	522	488	405	229
419	387	567	515	619	368	374	343	382	362	398	207	379	426	441	386	295	367	371	373	557	376	384	370	505	500	389	230	365	597	383	593	222	182	132	309	530	339	432	344	218	91	541	100000	212	303	120	482	397	260	533	191	251	462	369	428	148	89	229	319	289	296	173	227	290	532
273	254	374	337	417	213	219	169	220	218	236	228	224	223	267	228	174	211	220	237	356	218	224	217	300	310	238	211	217	386	244	381	51	183	121	99	318	134	221	132	149	191	333	212	100000	105	93	275	190	74	346	102	145	279	229	244	328	282	92	230	239	258	258	263	247	375
307	300	350	334	375	251	256	199	250	263	263	330	258	197	277	259	242	247	259	284	321	253	256	254	259	295	274	308	258	320	287	304	141	285	227	90	246	128	163	71	243	294	240	303	105	100000	189	180	98	43	241	131	121	174	146	140	394	358	91	181	212	234	296	277	223	274
323	296	454	407	502	267	273	234	279	266	295	186	278	308	334	284	205	265	272	280	440	274	282	270	387	388	291	187	267	477	290	473	106	143	64	189	410	220	312	224	143	110	425	120	93	189	100000	366	280	149	427	108	173	358	283	323	246	195	133	257	246	260	204	231	251	442
347	359	281	306	269	310	312	267	298	329	304	475	311	184	287	307	345	306	318	348	238	307	305	312	183	257	324	444	321	176	345	143	292	435	393	199	108	199	116	149	381	465	60	482	275	180	366	100000	85	223	121	307	277	111	215	111	571	538	266	293	342	361	464	432	352	216
301	305	287	290	298	255	258	206	247	271	256	395	258	151	251	256	276	251	263	293	251	253	253	258	188	245	273	367	265	228	291	208	208	354	308	119	153	130	92	68	302	380	144	397	190	98	280	85	100000	141	175	228	207	127	178	103	491	456	187	243	285	306	390	365	296	243
315	303	379	356	410	256	262	206	258	265	273	302	265	225	293	267	234	253	264	286	353	259	264	260	292	321	281	285	262	360	291	346	119	257	192	104	287	145	200	104	221	257	283	260	74	43	149	223	141	100000	278	89	95	210	157	175	351	315	52	171	191	212	257	244	201	301
465	475	396	426	374	425	428	379	415	443	422	565	427	303	407	424	451	421	433	463	352	422	421	428	301	377	441	539	436	268	461	224	376	523	467	293	215	305	235	242	474	535	122	533	346	241	427	121	175	278	100000	344	294	72	191	105	595	573	304	273	326	341	474	429	335	107
374	355	463	432	498	315	321	269	321	320	337	292	325	308	365	329	273	312	322	338	440	320	326	318	379	402	340	287	318	450	345	435	150	248	170	182	376	223	289	193	232	215	367	191	102	131	108	307	228	89	344	100000	65	272	180	239	264	232	42	149	143	161	168	163	150	343
408	393	470	450	496	349	354	299	352	356	367	354	358	317	388	360	318	345	356	377	442	352	357	352	379	414	373	346	354	435	382	414	196	309	233	198	360	240	283	191	288	280	335	251	145	121	173	277	207	95	294	65	100000	223	118	192	302	280	52	87	97	118	187	158	107	282
428	433	389	407	380	382	385	333	374	398	383	503	386	273	376	383	398	378	390	420	347	381	381	385	290	359	400	479	392	285	418	246	313	459	400	237	219	256	207	188	413	466	145	462	279	174	358	111	127	210	72	272	223	100000	126	35	524	502	233	209	261	278	405	362	271	116
452	446	465	463	472	396	401	344	394	409	407	455	403	323	415	404	387	393	405	430	428	398	400	400	366	420	419	440	404	390	433	358	276	410	338	236	317	270	269	203	377	393	263	369	229	146	283	215	178	157	191	180	118	126	100000	107	409	394	151	83	135	152	285	238	145	164
403	406	379	390	377	355	359	305	348	371	358	469	359	253	354	358	367	351	363	392	339	354	355	358	278	343	374	445	365	287	392	253	279	425	365	206	217	227	189	158	380	431	157	428	244	140	323	111	103	175	105	239	192	35	107	100000	494	470	200	188	239	257	377	337	250	143
563	532	699	653	745	509	515	480	522	505	538	351	521	551	579	527	440	508	514	517	683	517	525	512	628	633	532	377	508	709	528	697	351	329	276	426	637	461	545	450	364	237	630	148	328	394	246	571	491	351	595	264	302	524	409	494	100000	59	305	338	291	287	129	191	286	568
507	476	649	600	697	455	460	427	468	450	484	292	466	503	526	473	384	454	459	461	635	462	470	457	582	583	477	318	453	668	472	659	301	271	220	381	597	414	502	409	307	178	598	89	282	358	195	538	456	315	573	232	280	502	394	470	59	100000	274	330	288	288	137	200	285	556
358	342	431	406	462	298	304	250	302	305	317	307	307	277	341	311	266	295	306	325	406	302	307	301	345	372	323	297	303	410	331	395	144	262	188	153	337	195	251	156	237	243	325	229	92	91	133	266	187	52	304	42	52	233	151	200	305	274	100000	138	148	168	205	192	157	310
485	473	525	514	540	426	432	376	428	436	442	441	435	376	458	437	402	423	435	457	492	429	433	430	429	474	451	434	433	465	461	436	281	396	318	269	391	309	331	250	375	361	344	319	230	181	257	293	243	171	273	149	87	209	83	188	338	330	138	100000	53	69	210	157	62	230
505	490	561	545	581	446	451	396	449	453	464	432	455	409	484	458	413	442	453	473	530	449	454	449	467	507	470	430	451	511	479	484	290	388	309	295	436	336	369	283	376	342	394	289	239	212	246	342	285	191	326	143	97	261	135	239	291	288	148	53	100000	21	162	106	11	280
526	510	583	566	603	466	472	417	470	473	485	446	475	431	506	478	432	463	474	493	552	470	475	469	489	529	491	445	471	531	499	504	309	403	324	316	457	357	391	304	393	353	413	296	258	234	260	361	306	212	341	161	118	278	152	257	287	288	168	69	21	100000	157	97	11	290
524	499	630	595	666	465	471	426	475	466	491	366	476	476	524	481	408	463	471	482	608	471	478	468	548	567	490	380	467	616	491	598	297	332	258	349	542	389	457	361	345	255	522	173	258	296	204	464	390	257	474	168	187	405	285	377	129	137	205	210	162	157	100000	64	156	440
536	514	623	595	653	476	482	432	483	480	499	407	487	468	528	491	427	474	483	497	597	481	488	479	535	564	501	415	479	593	505	571	309	369	291	344	518	385	440	347	373	301	488	227	263	277	231	432	365	244	429	163	158	362	238	337	191	200	192	157	106	97	64	100000	98	386
514	499	572	555	592	455	461	406	458	462	473	437	464	420	494	467	421	452	462	482	541	459	464	458	478	518	480	435	460	522	488	495	298	394	314	305	447	346	380	293	382	345	405	290	247	223	251	352	296	201	335	150	107	271	145	250	286	285	157	62	11	11	156	98	100000	287
544	548	496	520	479	497	501	447	489	513	499	602	501	388	491	499	509	493	506	535	453	496	496	500	399	471	516	582	507	375	534	330	415	558	492	348	318	370	321	300	517	552	229	532	375	274	442	216	243	301	107	343	282	116	164	143	568	556	310	230	280	290	440	386	287	100000
>