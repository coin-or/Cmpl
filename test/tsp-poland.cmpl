%data tsp-poland.cdat
#%data tsp-poland-open.cdat
%arg -solver cplex
%display nonZeros
%opt cbc threads 8
%opt cbc ratio 0.02

parameters:
	#nrOfCities := len(cities);
	nrOfCities:=66;

variables:
	x[cities,cities]: binary;
	u[cities]: real[1..];

objectives:
	distance: sum{i in cities, j in cities: d[i,j]* x[i,j]} ->min;

constraints:
	sos_i {j in cities:  sum{i in cities: x[i,j]}=1; }
	sos_j {i in cities:  sum{j in cities: x[i,j]}=1; }

	noSubs {i in cities, j in cities, i!=j, j!=start: 
		u[i] - u[j] + nrOfCities * x[i,j] <= nrOfCities-1; }
	 