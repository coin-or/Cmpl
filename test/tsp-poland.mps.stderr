fatal error (module precompile: run): file error: cannot open file for input: cmpl input file 'tsp-poland.mps'; at ()
